torch/_inductor/pattern_matcher.py

```diff
@@ -1839,11 +1844,22 @@ def placeholder(
         def call_function(
             self, target: str, args: Sequence[Any], kwargs: Mapping[str, Any]  # type: ignore[override]
         ) -> PatternExpr:
-            args, kwargs = pytree.tree_map(process_arg, (args, kwargs))
+            # Indexing is critical for matching getitem nodes, so we can't ignore int args here
+            if target == operator.getitem and int in ignore_types:
+                temp_ignore_types = tuple(t for t in ignore_types if t is not int)
+
+                def process_arg_fun(
+                    x: T, ignore_types_override: Optional[Sequence[Type[Any]]] = None
+                ) -> Union[T, KeywordArg, Ignored]:
+                    return process_arg(x, ignore_types_override=temp_ignore_types)
```

Comment:
nit: could inline `ignore_types_override= tuple(t for t in ignore_types if t is not int)`

Reply:
done in https://github.com/pytorch/pytorch/pull/140193/commits/3d9df78029f4f21b91b04cc64655690760f314a8